external/fv3fit/fv3fit/reservoir/model.py

```diff
@@ -1,35 +1,43 @@
 import fsspec
 from fv3fit.reservoir.readout import ReservoirComputingReadout
-import numpy as np
 import os
+from typing import Optional, Iterable, Hashable
 import yaml
 
+from fv3fit import Predictor
+from .._shared import StandardScaler
 from .reservoir import Reservoir
+from .domain import RankDivider
+from fv3fit._shared import io
+from .utils import square_even_terms
+from .autoencoder import Autoencoder
 
 
-def _square_evens(v: np.ndarray) -> np.ndarray:
-    evens = v[::2]
-    odds = v[1::2]
-    c = np.empty((v.size,), dtype=v.dtype)
-    c[0::2] = evens ** 2
-    c[1::2] = odds
-    return c
+def _exists_in_dir(file_name, dir):
+    fs = fsspec.get_fs_token_paths(dir)[0]
+    contents = [os.path.basename(os.path.normpath(p)) for p in fs.ls(dir)]
+    return file_name in contents
 
 
-def _square_even_terms(v: np.ndarray, axis=1) -> np.ndarray:
-    return np.apply_along_axis(func1d=_square_evens, axis=axis, arr=v)
-
-
-class ReservoirComputingModel:
+@io.register("pure-reservoir")
+class ReservoirComputingModel(Predictor):
     _RESERVOIR_SUBDIR = "reservoir"
     _READOUT_SUBDIR = "readout"
     _METADATA_NAME = "metadata.yaml"
+    _SCALER_NAME = "scaler.npz"
+    _RANK_DIVIDER_NAME = "rank_divider.yaml"
+    _AUTOENCODER_SUBDIR = "autoencoder"
 
     def __init__(
         self,
+        input_variables: Iterable[Hashable],
+        output_variables: Iterable[Hashable],
```

Comment:
Any reason you use `Iterable` here but `Sequence` in the `ReservoirTrainingConfig`?

Reply:
Iterable here matches the parent `Predictor` class. I updated the config class to also use Iterable.